flink-runtime/src/main/java/org/apache/flink/runtime/zookeeper/filesystem/FileSystemStateStorageHelper.java

```diff
@@ -77,4 +83,18 @@ public FileSystemStateStorageHelper(Path rootPath, String prefix) throws IOExcep
 	private Path getNewFilePath() {
 		return new Path(rootPath, FileUtils.getRandomFilename(prefix));
 	}
+
+	@Override
+	public void closeAndCleanupAllData() {
+		try {
+			fs.delete(rootPath, true);
```

Comment:
We cannot simply delete the rootPath, because it's `HighAvailabilityOptions.HA_STORAGE_PATH`. Thus, we would also delete files from other components in this directory (e.g. the `BlobServer`). I think the `FileSystemStateStorageHelper` has to create a sub directory for the state to store. That directory, we can easily delete.